### PR TITLE
Add time travelling support

### PR DESCRIPTION
**Pull request by freund17**
closes https://github.com/alisd23/mst-react-router/issues/3

As I had the code forked anyways, I implemented https://github.com/alisd23/mst-react-router/issues/3 as well. (I, too, really missed this feature ;) )

This branch currently includes the changes made in https://github.com/alisd23/mst-react-router/pull/14.
I'll update it as soon as https://github.com/alisd23/mst-react-router/pull/14 is taken care of.

I just wanted to put this PR out there so you know that it's coming.

### CODE REVIEW

**Comment by alisd23** [comment]
@freund17 This looks cool. So I've merged in #14 now. Could you update this PR to match master? Then I'll do a review 👍  Thanks!

File: package.json [hunk 1]
```diff
@@ -69,7 +69,9 @@
     "webpack-cli": "^2.0.10"
   },
   "peerDependencies": {
-    "mobx-state-tree": "^3.0.0",
+    "mobx": "^4.3.1",
+    "mobx-state-tree": "^3.8.0",
+    "react": "^15.0.0",
```
Comment (alisd23) [inline package.json]: I don't think we need the explicit `React` dependency. This peer dependency is implied through `react-router`, and we don't actually depend on `React` explicitly in the code.

File: webpack.config.js [hunk 2]
```diff
@@ -42,13 +42,37 @@ module.exports = {
...
+    },
     {
       'mobx-state-tree': {
         root: 'MobxStateTree',
         commonjs2: 'mobx-state-tree',
         commonjs: 'mobx-state-tree',
         amd: 'mobx-state-tree'
       }
+    },
+    {
```
Comment (alisd23) [inline webpack.config.js]: Any reason for adding these (React, ReactRouter) to the config? Only libraries which we explicitly depend on and need to reference in the code should need to be listed here.

**Comment by alisd23** [comment]
> What we actually should be depending on is history

Good point, I guess the `peerDependencies` you have listed now in this PR are the most correct. 

> The minimum versions of the peerDependencies can probably be pushed quite a bit lower
Also agree if possible. Don't want people to suddenly have a bunch of version warnings. However they are only warnings and we'd have to laboriously figure out the minimum versions, so I'm not too bothered.

> * just to be clear: renaming the library seems like a terrible idea

Phew, yep let's not.

I'm not actually using this library in a project at the moment, so it's great that you are and can test this change in practice. Happy to wait until you've tried it out a bit before merging, as it's quite a key change.

Thanks 👍

**Comment by freund17** [comment]
I tested it a little and feel confident, that everything works fine.

A few gotchas though:
- when time-travelling, the history will always be `replaced`
- `_setHistory` no longer sets `routerModel.location`
- using `location.key` as equality criteria, does not handle locations without `key`\*
- I do not use yarn, so `yarn.lock` is out of date

---

\* Locations without `key` are always seen as different.
This means, when setting a location without a `key` directly in the store (in general not possible), it would:
1. flush to the history-api (as expected)
2. be given a key from the history api
3. be flushed back to the store

The result is, that a trigger listening on `store.location` would trigger twice.

To me this seems like a negligible side-effect in a rare corner-case, which has an obvious work-around (just not setting `store.location` directly and using the provided functions instead).
I just wanted to list it here for future reference.

**Comment by alisd23** [comment]
Yep, agreed I think that edge-case will be uncommon enough to not worry about.

I'll get the yarn.lock updated then I'm happy to merge and release as a new minor version: `2.3.0`.

**Pull request merged**